### PR TITLE
Roomify frontend

### PR DESCRIPTION
**Pull request by Daf98**
[Link to backend PR](https://github.com/andgarzonmal/capstone-project-back-end/pull/32)

#### Basics
- You should follow the layout of [the provided design](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign). You should change only the titles, descriptions, and photos - in order to create a website about something other than motorcycles.
- Select a theme for your website - is it going to be a website for booking doctor appointments, booking online classes, or something else?

#### Features

**Core features - for teams of each size**
- The user logs in to the website, only by typing the username (a proper authenticated login is a requirement if your group is made of 5 people).
- In the navigation panel, the user can see links to:
    - Motorcycles/doctors/classes/items that you selected as a theme.
    - "Reserve" form.
    - "My reservations".
    - "Add motorcycle/doctor/class/item that you selected as a theme" (the link is visible to everybody).
    - "Delete motorcycle/doctor/class/item that you selected as a theme" (the link is visible to everybody).
- On the main page, the user can see a [list of motorcycles/doctors/classes/items that you selected as a theme](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005577).
- When the user selects a specific item, they can see the [details page](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005579) with its full description (skip the "Rotate image" button).
    - In the details page, the user can click the "Reserve" button (in the design you can see the "Configure" button - please replace it with the "Reserve" button).

- When the user clicks the "Add item" link in the navigation panel they can see a form for adding a new item.

- Make the app responsive, creating both mobile and desktop versions.

**If your team has 3-4 members these are the required features; otherwise, consider them as optional features.**
- When the user clicks the "Delete item" link in the navigation panel they can see a list of all items with title and "Delete" button.
    - When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.
- To reserve an appointment, the user has to select a date and city (username and selected item are autofilled).
    - Use the design based on the ["Book a vespa test-ride"](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005583) and add all necessary inputs.
    - The user can also access the "Reserve" page from the navigation panel. In that case only username is autofilled.
- When the user clicks the "My reservations" link in the navigation panel they can see a list of their reservations (with information about item name, date and city).
- Add full documentation for your API.

**If your team has 5 members these are the required features; otherwise, consider them as optional features.**

- Implement proper user authentication from the front-end to the server.
- Make sure that the "Add item" and "Delete item" links are accessible only by users who are admins.

#### Technical set up
- Set up the repository/repositories on GitHub and use Gitflow.
- You should use Postgres as your database
- Use Rails to create backend API.
- Use React & Redux to create frontend UI.
- You can choose if you want to set up your project as two separate apps or as one.
- The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

#### Unexpected issues
In the case of some unexpected issues that can cause delay, you might need to compromise some features. Remember that if you implement at least 80% of the required features and provide a good explanation for the fact that some requirements are not met, your project will be accepted.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,16 +7,22 @@
     "@emotion/styled": "^11.10.4",
     "@mui/icons-material": "^5.10.9",
     "@mui/material": "^5.10.10",
+    "@reduxjs/toolkit": "^1.8.6",
```
Comment (Meltrust) [inline package.json:10]: On your readme:

- [x] Your readme does not correspond to the Microverse Readme Template (your backend does!).  Please, go [here](https://github.com/microverseinc/readme-template/blob/master/README.md) and use this template to craft your professional readme.  Please, don't forget the screenshot!  Your design is beautiful let's show it off 💪

Comment (Meltrust) [inline package.json:10]: - [x] Your frontend proyect has no tests. Kindly add at least some tests to ensure your app's integrity. Please, remember also to include instructions about how to run the tests on your readme.

In the industry, we write tests every day for each feature implemented. You will need this skill for your next brand-new job 💪

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import Login from './pages/Login';
+import SignUp from './pages/SignUp';
+import CreateRoom from './pages/CreateRoom';
+import PrivateRoute from './components/PrivateRoute';
+import MyReservations from './components/MyReservations';
+import { displayRooms } from './redux/actionCreator';
+import DeleteRoom from './pages/DeleteRoom';
+import ReserveRoom from './components/ReserveRoom';
+
+function App() {
```
Comment (Meltrust) [inline src/App.js]: - [x] You are using the `function` keyword for many of your functions.  Kindly refactor them to ES6 arrow functions.

## Explanation

In the industry, the `function` keyword has been soft deprecated almost everywhere in favor of ES6.  On your next job, you'll be most likely required to write ES6.

## Tips and extra help

Just perform a search on Vscode for the `function` word.  Refactor to arrow function as you find them (will take you about 5-10 minutes at most, don't worry).

That way, your project will comply with the latest coding standards 💪

File: package.json (hunk 1 above)
Comment (Meltrust) [inline package.json:10]: - [x] Kindly deploy your app and add a link to it to your readme.  Your app is really beautiful.  Let's show it off to the world! (and potential employers, of course).

That way, your users won't have to install your app locally in order to check it out, and potential employers will be more likely to take a look. 💪

Comment (Meltrust) [inline package.json:10]: - [x] Your app has many eslint errors and react warnings.  So much, it's impossible to run your frontend app without deactivating ESlint.

Please, kindly fix those eslint and React errors in order to make your app work as intended. 👍

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+)}
+              />
+            ))}
+
+            <Route exact path="/" element={<PrivateRoute />}>
+              <Route exact path="/create-room" element={<CreateRoom />} />
+              <Route exact path="/delete-room" element={<DeleteRoom />} />
+            </Route>
+            {/* {rooms.map((room) => <Route key={room.name}
+            exact path={`/rooms/${room.id}/details`} element={<Details />} />)} */}
```
Comment (Meltrust) [inline src/App.js]: - [x] There are a lot of old pieces of commented code like this over your app's files.  Please, remove it to improve the cleanliness of your app. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @Daf98,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  The app is visually beautiful✔️
-  App is working (only if I deactivate ESlint) ✔️ 
 

You can do it

## Suggested Changes ♻️
_Check the comments under the review._

You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.


Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

File: package.json (hunk 1 above)
Comment (Daf98) [reply package.json:10]: @Meltrust We have checked all the project requirements and we haven't been able to see testing as a requirement for the frontend. Please let us know where these tests are required.

Comment (Meltrust) [reply package.json:10]: @Daf98 as mentioned in the review summary:

image

Therefore, in this particular project and regarding this review cycle, nothing is a hard requirement. 

This particular special review cycle is meant to prepare you for your final presentation, according to real-world industry standards.

**Review by Meltrust — APPROVED** [review]
#### Hi @Daf98,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Commented code has been removed✔️ 
- Linter errors have been resolved✔️ 
- The deployed app works well ✔️
- You are now using ES6 functions ✔️ 
- Readme has been corrected ✔️ 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏


------

**Pull request merged**